Karl/karl.h

```diff
@@ -405,7 +560,17 @@ class GenericKarl final {
                          }
                          return Response("OK\n");
                        },
-                       std::move(r)).Detach();
+                       std::move(r)).Wait();
+          {
+            std::lock_guard<std::mutex> lock(services_keepalive_cache_mutex_);
+            if (services_keepalive_time_cache_.count(codename) == 0u) {
```

Comment:
As you initialize it below, unless `0u` is a valid value, might as well do `auto& placeholder = services_keepalive_time_cache_[codename];` and compare it to zero.